src/SamplesApp/SamplesApp.UnitTests.Shared/Controls/UnitTest/UnitTestMethodInfo.cs

```diff
@@ -91,28 +80,50 @@ public UnitTestMethodInfo(object testClassInstance, MethodInfo method)
 	private bool HasCustomAttribute<T>(MemberInfo? testMethod)
 		=> testMethod?.GetCustomAttribute(typeof(T)) != null;
 
-	public bool IsIgnored(out string ignoreMessage)
+	// Adopted from https://github.com/microsoft/testfx/blob/47dee826a0a3eb7a2d9d089ed8aba9d2dabfe82e/src/Adapter/MSTest.TestAdapter/Helpers/AttributeHelpers.cs
+	private static bool IsIgnored(ICustomAttributeProvider member, out string? ignoreMessage)
 	{
-		var ignoreAttribute = Method.GetCustomAttribute<IgnoreAttribute>();
-		if (ignoreAttribute == null)
+		var attributes = member.GetCustomAttributes(typeof(ConditionBaseAttribute), inherit: false).Cast<ConditionBaseAttribute>();
+		IEnumerable<IGrouping<string, ConditionBaseAttribute>> groups = attributes.GroupBy(attr => attr.GroupName);
+		foreach (IGrouping<string, ConditionBaseAttribute>? group in groups)
 		{
-			ignoreAttribute = Method.DeclaringType?.GetCustomAttribute<IgnoreAttribute>();
+			bool atLeastOneInGroupIsSatisfied = false;
+			string? firstNonSatisfiedMatch = null;
+			foreach (ConditionBaseAttribute attribute in group)
+			{
+				// Mode property is internal in MSTest, thus we cannot use it here.
```

Comment:
TODO: This currently won't compile due to https://github.com/microsoft/testfx/issues/5581